### DISCUSSION TITLE
Order in the Edge attributes list

### DISCUSSION THREAD

**Discussion by antonioaa1979** [danielegrattarola/spektral#255] [Q&A] [answered]
Hi all,

as per documentation, the Edge matrix of a graph can be represented in a sparse format as arrays of shape (n_edges, n_edge_features), which is obviously the convenient way in the case of giant graphs. This is a list of every edge and the corresponding attributes. Unfortunately in the documentation it doesn't mention clearly what the order of this list should be and how to keep the connection with the adjacency matrix. Any hints?

Thanks,
Antonio

**Comment by danielegrattarola** [comment, marked as answer]
Hi, 

you are right, this should be more clearly stated in the docs. 
The correct order for the edge list is obtained by sorting the indices of the adjacency matrix in COO format in lexicographic order, for example: 

0, 1
0, 2
1, 0
2, 0

For a given numpy or scipy matrix, you can find the correct indices/order with 
```py
>>> from scipy import sparse
>>> a = ...  # adjacency matrix
>>> sparse.find(a)  # returns sorted indices and entries
```

Cheers

**Reply by antonioaa1979** [reply]
Thanks Daniele, that's what i was guessing too, but thanks for the confirmation
